### PR TITLE
swarm/pss: Add GetPeerTopics, GetPeerAddress api method

### PR DESCRIPTION
**Pull request by nolash**
This PR adds pss method and api call to check existence of public key peer records in node.

### CODE REVIEW

File: swarm/pss/api.go [hunk 1]
```diff
@@ -137,3 +137,13 @@ func (pssapi *API) SendAsym(pubkeyhex string, topic Topic, msg hexutil.Bytes) er
 func (pssapi *API) SendSym(symkeyhex string, topic Topic, msg hexutil.Bytes) error {
 	return pssapi.Pss.SendSym(symkeyhex, topic, msg[:])
 }
+
+func (pssapi *API) GetPeer(pubkeyhex string) ([]Topic, error) {
```
Comment (acud) [inline swarm/pss/api.go]: if you are returning topics, why is the method called `GetPeer`?
rename to `GetPeerTopics` or `GetTopicsForPeer`

Comment (acud) [reply swarm/pss/api.go]: also, why not change the signature so you could just return the `pssapi.Pss.GetPeer(pubkeyhex)` directly? (as in `GetPeerAddress`)

Comment (nolash) [reply swarm/pss/api.go]: I think you can only return one param with RPC. Good idea about method name though, will rename.

File: swarm/pss/pss.go [hunk 2]
```diff
@@ -518,6 +518,29 @@ func (self *Pss) GetSymmetricKey(symkeyid string) ([]byte, error) {
 	return symkey, nil
 }
 
+// Returns all recorded topic and address combination for a specific public key
+func (self *Pss) GetPeer(keyid string) (topic []Topic, address []PssAddress, err error) {
```
Comment (gbalint) [inline swarm/pss/pss.go]: If you renamed `api.GetPeerTopics` imho this should be renamed too (this does not return the peer neither)

File: swarm/pss/pss.go [hunk 3]
```diff
@@ -518,6 +518,29 @@ func (self *Pss) GetSymmetricKey(symkeyid string) ([]byte, error) {
...
+	self.pubKeyPoolMu.Lock()
```
Comment (gbalint) [inline swarm/pss/pss.go]: It is enough to RLock here

File: swarm/pss/pss.go [hunk 4]
```diff
@@ -518,6 +518,29 @@ func (self *Pss) GetSymmetricKey(symkeyid string) ([]byte, error) {
...
+	for t, p := range self.pubKeyPool[keyid] {
+		topic = append(topic, t)
+		address = append(address, *p.address)
+	}
+
+	return topic, address, nil
+}
+
+func (self *Pss) getPeerAddress(keyid string, topic Topic) (PssAddress, error) {
+	self.pubKeyPoolMu.Lock()
```
Comment (gbalint) [inline swarm/pss/pss.go]: Here it is also enough to RLock

File: swarm/pss/pss_test.go [hunk 5]
```diff
@@ -457,6 +457,76 @@ func TestKeys(t *testing.T) {
 	}
 }
 
+func TestGetPeer(t *testing.T) {
```
Comment (gbalint) [inline swarm/pss/pss_test.go]: There is no more GetPeer, so TestGetPeer is not the best name.... as I see it gets the topics and the address too... maybe `TestGetPeerData`?

File: swarm/pss/pss_test.go [hunk 6]
```diff
@@ -457,6 +457,76 @@ func TestKeys(t *testing.T) {
 	}
 }
 
+func TestGetPublickeyEntries(t *testing.T) {
```
Comment (zelig) [inline swarm/pss/pss_test.go:460]: why do we need a simulation to test this simple API call? unit test on pssapi like all other api methods ?

Comment (nolash) [reply swarm/pss/pss_test.go:460]: changed @zelig

**Review by gbalint — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**